modules/git/repo_compare.go

```diff
@@ -263,3 +270,43 @@ func (repo *Repository) GetDiffFromMergeBase(base, head string, w io.Writer) err
 	}
 	return err
 }
+
+// ReadPatch will check if a diff patch exists and return stats
+func (repo *Repository) ReadPatch(prID int64) (commitSHA string, err error) {
```

Comment:
Maybe we could change the function name

Reply:
I modified to ReadPatchCommit (seemed more logical than ReadPatchFirstLine, but I can still change to anything better